packages/backend-app-api/src/config/config.ts

```diff
@@ -30,19 +30,56 @@ import type { Config, AppConfig } from '@backstage/config';
 import { getPackages } from '@manypkg/get-packages';
 import { ObservableConfigProxy } from './ObservableConfigProxy';
 import { isValidUrl } from '../lib/urls';
+import path from 'path';
+import fs from 'fs';
+
+// Looks for a package.json with a workspace config to identify the root of the monorepo
+function findRootPath(searchDir: string): string {
+  let currentPath = searchDir;
+
+  // Some confidence check to avoid infinite loop
+  for (let i = 0; i < 1000; i++) {
+    const packagePath = resolvePath(currentPath, 'package.json');
+    const exists = fs.existsSync(packagePath);
+    if (exists) {
+      return currentPath;
+    }
+
+    let newPath = path.dirname(currentPath);
+    if (newPath === currentPath) {
+      newPath = path.join(currentPath, '..');
```

Comment:
Really? I would have thought that this would be an error condition (likely because hitting the root) right? Especially if you start at a normalized path.

I also see that `path.posix.dirname('C:\\a\\b')` returns `'.'`, so maybe that's also a corner case end condition to look out for - again, if you start at a normalized absolute path.

Reply:
I was seeing this with `dirname('/path/to/dir/')` which would return `/path/to/dir/`, ie dirname of a directory is itself.

Reply:
Updated, I think I was doing something weird here. Added test cases to assert the behavior too.